### ISSUE TITLE
DeviceDetails.id (PCD2) should be nullable.

### ISSUE THREAD

**Issue by maratal** [ably/specification#180]
Currently it is declared as non-nil for some reason:

```
class DeviceDetails: // PCD*
  id: String // PCD2
```

which leads to the bunch of problems and inconsistencies in the SDK(s) implementations.

**Comment by lawrence-forooghian** [comment]
Alternatively we can update the spec so that it reflects the behaviour implemented in https://github.com/ably/ably-cocoa/pull/1847, where a new device ID is generated as soon as the previous ones are cleared, so that `id` is never null.

**Comment by maratal** [comment]
> Alternatively we can update the spec so that it reflects the behaviour implemented in [ably/ably-cocoa#1847](https://github.com/ably/ably-cocoa/pull/1847), where a new device ID is generated as soon as the previous ones are cleared, so that `id` is never null.

It's an idea. But I think it's better to eventually make nullable `id` nevertheless. And schedule this change together with the v2 protocol. WDYT?

**Comment by lawrence-forooghian** [comment]
> But I think it's better to eventually make nullable id nevertheless.

Why do you think it better?

> And schedule this change together with the v2 protocol. WDYT?

What's the relation with the v2 protocol?

**Comment by maratal** [comment]
>Why do you think it better?

Hmm, good question. At least it's nullable in other SDKs already.

>What's the relation with the v2 protocol?

I mean v2 release is a breaking change and release of another breaking change together would make sense I think.

**Comment by maratal** [comment]
>The changes introduced by [version 2.0.0 of the specification](https://github.com/ably/specification/releases/tag/v2.0.0), which is the version that switched to using protocol v2, do not include any breaking API changes.

You are right, the one that was deleted is marked as deprecated. Ok, will change the specs then.
